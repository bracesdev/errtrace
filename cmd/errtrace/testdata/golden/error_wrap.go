//go:build ignore

package foo

type innerError struct{}

func (*innerError) Error() string {
	return "sadness"
}

type errorWrapper struct {
	Err error
}

func (e *errorWrapper) Error() string {
	return e.Err.Error()
}

// Unwrap should not be wrapped by errtrace.
func (e *errorWrapper) Unwrap() error {
	return e.Err
}

func Try() error {
	return &errorWrapper{Err: &innerError{}}
}
