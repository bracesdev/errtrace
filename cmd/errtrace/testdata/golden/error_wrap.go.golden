//go:build ignore

package foo; import "braces.dev/errtrace"

type innerError struct{}

func (*innerError) Error() string {
	return "sadness"
}

type errorWrapper struct {
	Err error
}

func (e *errorWrapper) Error() string {
	return e.Err.Error()
}

// Unwrap should not be wrapped by errtrace.
func (e *errorWrapper) Unwrap() error {
	return e.Err
}

func Try() error {
	return errtrace.Wrap(&errorWrapper{Err: &innerError{}})
}
